apps/code-examples/src/app/code-examples/flyout/flyout/custom-headers/demo.component.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { SkyFlyoutHarness } from '@skyux/flyout/testing';
+
+import { DemoComponent } from './demo.component';
+
+describe('Flyout harness', () => {
+  async function setupTest(option?: string): Promise<{
+    flyoutHarness: SkyFlyoutHarness;
+    fixture: ComponentFixture<DemoComponent>;
+  }> {
+    await TestBed.configureTestingModule({
+      imports: [DemoComponent, NoopAnimationsModule],
+    }).compileComponents();
+
+    const fixture = TestBed.createComponent(DemoComponent);
+    const loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+
+    switch (option) {
+      case 'url':
+        fixture.componentInstance.openFlyoutWithUrlPermalink();
+        break;
+      case 'action':
+        fixture.componentInstance.openFlyoutWithPrimaryAction();
+        break;
+      default:
+        fixture.componentInstance.openFlyoutWithIterators();
+        break;
+    }
+
+    const flyoutHarness: SkyFlyoutHarness =
+      await loader.getHarness(SkyFlyoutHarness);
+
+    return { flyoutHarness, fixture };
+  }
+
+  it('should set up the flyout with iterators', async () => {
+    const { flyoutHarness, fixture } = await setupTest();
+
+    await expectAsync(flyoutHarness.getAriaLabelledBy()).toBeResolvedTo(
+      'flyout-title',
+    );
+    await expectAsync(flyoutHarness.getAriaRole()).toBeResolvedTo('dialog');
+
+    expect(fixture.componentInstance.recordNumber).toBe(0);
+    await flyoutHarness.clickNextIteratorButton();
+    expect(fixture.componentInstance.recordNumber).toBe(1);
+    await flyoutHarness.clickPreviousIteratorButton();
+    expect(fixture.componentInstance.recordNumber).toBe(0);
+    // previous button is disabled
+    await flyoutHarness.clickPreviousIteratorButton();
```

Comment:
can we throw an error for this scenario where they attempt to click a disabled iterator?

Reply:
done